### PR TITLE
clang-tidy cleanups

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

**Comment by dota17** [comment]
It's good!

File: src/directives.cpp [hunk 1]
```diff
@@ -1,11 +1,11 @@
 #include "directives.h"
 
 namespace YAML {
-Directives::Directives() : version{true, 1, 2}, tags{} {}
```
Comment (jbeder) [inline src/directives.cpp:4]: Prefer to keep these explicit initialize statements (throughout).

File: src/emitterstate.cpp [hunk 2]
```diff
@@ -94,15 +89,13 @@ EmitterNodeType::value EmitterState::NextGroupType(
   if (type == GroupType::Seq) {
     if (GetFlowType(type) == Block)
       return EmitterNodeType::BlockSeq;
-    else
```
Comment (jbeder) [inline src/emitterstate.cpp:97]: This is a 2x2 grid of results, and probably worth keeping as-is.

File: src/node_data.cpp [hunk 3]
```diff
@@ -209,12 +206,9 @@ node* node_data::get(node& key, shared_memory_holder /* pMemory */) const {
     return nullptr;
   }
 
-  for (node_map::const_iterator it = m_map.begin(); it != m_map.end(); ++it) {
-    if (it->first->is(key))
-      return it->second;
-  }
-
-  return nullptr;
+  auto it = std::find_if(m_map.begin(), m_map.end(), [&](const std::pair<node*, node*> node)
```
Comment (jbeder) [inline src/node_data.cpp:209]: I think this still reads better as a for loop, but perhaps you can change it to:

for (const auto& kv : m_map) {
  if (kv.first->is(key)) {
    return it->second;
  }
}
return nullptr;

File: src/node_data.cpp [hunk 4]
```diff
@@ -230,9 +224,10 @@ node& node_data::get(node& key, shared_memory_holder pMemory) {
       throw BadSubscript(m_mark, key);
   }
 
-  for (node_map::const_iterator it = m_map.begin(); it != m_map.end(); ++it) {
-    if (it->first->is(key))
-      return *it->second;
+  auto it = std::find_if(m_map.begin(), m_map.end(), [&](const std::pair<node*, node*> node)
```
Comment (jbeder) [inline src/node_data.cpp:227]: same here

File: src/nodeevents.cpp [hunk 5]
```diff
@@ -32,8 +32,8 @@ void NodeEvents::Setup(const detail::node& node) {
     return;
 
   if (node.type() == NodeType::Sequence) {
-    for (detail::const_node_iterator it = node.begin(); it != node.end(); ++it)
-      Setup(**it);
+    for (auto && it : node)
```
Comment (jbeder) [inline src/nodeevents.cpp:35]: Why && instead of const&?

File: src/nodeevents.cpp [hunk 6]
```diff
@@ -77,9 +77,8 @@ void NodeEvents::Emit(const detail::node& node, EventHandler& handler,
       break;
     case NodeType::Sequence:
       handler.OnSequenceStart(Mark(), node.tag(), anchor, node.style());
-      for (detail::const_node_iterator it = node.begin(); it != node.end();
-           ++it)
-        Emit(**it, handler, am);
+      for (auto && it : node)
```
Comment (jbeder) [inline src/nodeevents.cpp:80]: ditto

File: src/nodeevents.cpp (hunk 5 above)
Comment (neheb) [reply src/nodeevents.cpp:35]: No idea. clang-tidy did this.

File: src/node_data.cpp (hunk 3 above)
Comment (neheb) [reply src/node_data.cpp:209]: Sure. Using algorithms seems to be preferred by some. They result in better code generation sometimes. I’ve seen that with std::accumulate.

**Comment by neheb** [comment]
Ran all commits through git clang-format.

**Comment by neheb** [comment]
Removed member init commit.

File: include/yaml-cpp/node/detail/impl.h [hunk 7]
```diff
@@ -125,13 +127,11 @@ inline node* node_data::get(const Key& key,
       throw BadSubscript(m_mark, key);
   }
 
-  for (node_map::const_iterator it = m_map.begin(); it != m_map.end(); ++it) {
-    if (it->first->equals(key, pMemory)) {
-      return it->second;
-    }
-  }
+  auto it = std::find_if(m_map.begin(), m_map.end(), [&](const kv_pair m) {
```
Comment (jbeder) [inline include/yaml-cpp/node/detail/impl.h:130]: I still prefer the for loop.

File: include/yaml-cpp/node/convert.h [hunk 8]
```diff
@@ -255,9 +252,7 @@ template <typename T, typename A>
 struct convert<std::list<T,A>> {
   static Node encode(const std::list<T,A>& rhs) {
     Node node(NodeType::Sequence);
-    for (typename std::list<T,A>::const_iterator it = rhs.begin();
-         it != rhs.end(); ++it)
-      node.push_back(*it);
+    std::copy(rhs.begin(), rhs.end(), std::back_inserter(rhs));
```
[the remark this answers is not in the document]
Comment (neheb) [reply include/yaml-cpp/node/convert.h:255]: I did. That's embarrassing.

**Pull request closed without merging**